Directory.Build.props

```diff
@@ -68,7 +68,7 @@
         - src/tasks/MobileBuildTasks/Apple/AppleProject.cs
         - https://github.com/dotnet/sdk repo > src/Installer/redist-installer/targets/GeneratePKG.targets
      -->
-    <AndroidApiLevelMin>21</AndroidApiLevelMin>
+    <AndroidApiLevelMin>29</AndroidApiLevelMin>
```

Comment:
this needs to be conditioned on CoreCLR so we don't change the Mono build. same in eng/native/build-commons.sh

Reply:
Should this be 28? @janvorli tested it recently with 28 https://github.com/dotnet/arcade/blob/88a10ba00ed798ee289d786db9f6339acd682a99/eng/common/cross/build-android-rootfs.sh#L24. We can provide a fallback for that glob API to further lower the requirement.

Reply:
I think 29 is necessary because that's the first that does not use emulated TLS.

Reply:
We can add a run-time check for emulated TLS selection and still lower the requirement.

Reply:
Did we add a workaround for that? When I ran things previously we only had that for NativeAOT. Assuming that's true, the build will fail.

Reply:
Emulated TLS works on linux-bionic and has been for a while, so in theory it should work on older Androids, newer ones can use that. I have recently dealt with Emulated TLS on riscv64 (and later switched to real TLS but still having some issues https://github.com/dotnet/runtime/pull/111317#issuecomment-2589173720). We can tweak things in asm code to select which path to take `GETTHREAD_ETLS_1` vs. `INLINE_GETTHREAD`. Right now it's under the build condition `FEATURE_EMULATED_TLS`.

Reply:
Emulated TLS works on linux-bionic and has been for a while, so in theory it should work on older Androids, newer ones can use the real TLS. I have recently dealt with Emulated TLS on riscv64 (and later switched to real TLS but still having some issues https://github.com/dotnet/runtime/pull/111317#issuecomment-2589173720). We can tweak things in asm code to select which path to take `GETTHREAD_ETLS_1` vs. `INLINE_GETTHREAD`. Right now it's under the build condition `FEATURE_EMULATED_TLS`.

Reply:
It appears that there is going to be some amount of work to get this right < API level 29. My preference would be to do any of that work in a follow up.